### PR TITLE
Bump to latest github.com/SKF/go-utility/v2

### PR DESCRIPTION
**Pull request by j-sv**
This version contains an upgrade of the github.com/golang-jwt package.

### CODE REVIEW

File: authentication/handler.go [hunk 1]
```diff
@@ -151,23 +148,18 @@ func (m *Middleware) decorateValidRequest(ctx context.Context, r *http.Request,
...
-			return custom_problems.UnverifiableToken()
-		case ve.Errors&jwt_go.ValidationErrorSignatureInvalid != 0:
-			return custom_problems.UnverifiableToken()
-		case ve.Errors&jwt_go.ValidationErrorExpired != 0:
-			return custom_problems.ExpiredToken()
-		case ve.Errors&jwt_go.ValidationErrorNotValidYet != 0:
-			return custom_problems.NotYetValidToken()
-		}
-
-		return custom_problems.InvalidToken(ve.Error())
```
Comment (Ekberg) [inline authentication/handler.go:170]: Does the JWT library expose any new validation errors? Perhaps that was handled by this case in the old version.

File: authentication/handler.go [hunk 2]
```diff
@@ -151,23 +148,18 @@ func (m *Middleware) decorateValidRequest(ctx context.Context, r *http.Request,
...
+		return custom_problems.UnverifiableToken()
+	case errors.Is(err, jwt_go.ErrTokenSignatureInvalid):
+		return custom_problems.UnverifiableToken()
+	case errors.Is(err, jwt_go.ErrTokenExpired):
+		return custom_problems.ExpiredToken()
+	case errors.Is(err, jwt_go.ErrTokenNotValidYet):
+		return custom_problems.NotYetValidToken()
 	}
 
 	if strings.HasPrefix(err.Error(), "token is not valid") {
```
Comment (Ekberg) [inline authentication/handler.go:165]: Are these "custom" mappings still valid in v5?

File: authentication/handler.go [hunk 3]
```diff
@@ -176,7 +168,7 @@ func jwtErrorToProblem(err error) error {
 
 	if strings.HasPrefix(err.Error(), "parse with claims failed:") ||
 		strings.HasPrefix(err.Error(), "failed to validate claims:") {
-		return custom_problems.InvalidToken(old_errors.Cause(err).Error())
+		return custom_problems.InvalidToken(errors.Unwrap(err).Error())
```
Comment (Ekberg) [inline authentication/handler.go:171]: Is this handled in the same way in v5? I think the `.Cause` call was needed for some internal reason. Guessing we don't have tests here?

File: authentication/handler_test.go [hunk 4]
```diff
@@ -1,20 +1,22 @@
...
-	err := jwt_go.NewValidationError("bad token", jwt_go.ValidationErrorExpired)
+	err := fmt.Errorf("bad token: %w", jwt_go.ErrTokenExpired)
 
 	problem := jwtErrorToProblem(err)
 
 	pp, ok := problem.(problems.Problem)
 	require.True(t, ok)
 
-	require.Equal(t, "/problems/expired-authentication-token", pp.ProblemType())
+	assert.Equal(t, "/problems/expired-authentication-token", pp.ProblemType())
```
Comment (Ekberg) [inline authentication/handler_test.go:21]: Why did we need to add assert here?

File: authentication/handler.go (hunk 3 above)
Comment (j-sv) [reply authentication/handler.go]: I guess it was just due to the old `github.com/SKF/v2/jwt` returning `pkg/errors` errors, which was changed in the latest version...

**Review by Ekberg — APPROVED** [review]

**Pull request merged**